File: scripts/ScriptHandler.js
var ChannelDuplexHandler = Java.type("io.netty.channel.ChannelDuplexHandler");
var MyChannelDuplexHandler = Java.extend(ChannelDuplexHandler, {
    channelRead: function(ctx, msg) {
        ctx.fireUserEventTriggered("Hello from Javascript");
        ctx.fireChannelRead(msg);
    }
});
var _ = new MyChannelDuplexHandler();

File: scripts/ScriptHandler.py
from io.netty.channel import ChannelDuplexHandler, ChannelHandler
from io.netty.channel import ChannelHandlerContext
from io.netty.channel import ChannelPromise

class JythonHandler(ChannelDuplexHandler, ChannelHandler):
    def __init__(self):
        pass

    def channelRead(self, ctx, msg):
        ctx.fireUserEventTriggered("Read from Jython")
        ctx.fireChannelRead(msg)

    def write(self, ctx, msg, promise):
        ctx.fireUserEventTriggered("Write from Jython")
        ctx.write(msg, promise)

_ = JythonHandler()

